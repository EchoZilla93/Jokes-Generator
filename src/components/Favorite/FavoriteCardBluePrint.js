import React, {useState } from 'react';
import PropTypes from 'prop-types';
import { getFavoriteJokes } from '../../localStorage';

import './favoriteCard.scss';

export const FavoriteCardBluePrint = () => {
    const [favoriteJokes, setFavoriteJokes] = useState([]);
    const renderFavorite = favoriteJokes && favoriteJokes.map(joke => {
        return (
        <div className="favoriteCard">
            <div className="favoriteCard__status">
                <i className="far fa-heart heart"></i>
            </div>
            <div className="favoriteCard__staticImage">
                <i className="far fa-comment-dots static"></i>
                </div>
            <div className="favoriteCard__mainContent">
                <h6 className="favoriteCard__ID">ID: <a href='/'>{joke.id}</a></h6>
            <div className="textcontent">
                 <p>{joke.favJoke}</p>
                </div>
            </div>
            <div className="favoriteCard__updateInfo">
        <p className="updateInfo">Last Update: {joke.updated_at}hours ago</p>
            </div>
            </div>
            )})

    return (
        <div>
            {renderFavorite}
        </div>
        
    );
};

FavoriteCardBluePrint.propType = {
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    updateInfo: PropTypes.number.isRequired
};

export default FavoriteCardBluePrint;