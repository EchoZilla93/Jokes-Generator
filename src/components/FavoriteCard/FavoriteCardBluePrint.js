import React, { Fragment } from 'react';
import { useSelector, useDispatch } from 'react-redux'
import { removeFavorite } from '../../actions';

import './favoriteCard.scss';

export const FavoriteCardBluePrint = () => {
    const dispatch = useDispatch();
    const favoriteJokes = useSelector(state => state.jokes.favoriteJokes);

    return (
        <Fragment>
            {favoriteJokes &&
                favoriteJokes.map(joke => (
                    <div key={joke.id}>
                        <div className="favoriteCard">
                            <div className="favoriteCard__status">
                                <i onClick={() => dispatch(removeFavorite(joke.id))} className="fas fa-heart" id="icon"></i>
                            </div>
                            <div className="favoriteCard__staticImage">
                                <i className="far fa-comment-dots static"></i>
                            </div>
                            <div className="favoriteCard__mainContent">
                                <h6 className="favoriteCard__ID">ID: <a href={joke.url}>{joke.id}</a></h6>
                                <div className="textcontent">
                                    <p>{joke.value}</p>
                                </div>
                            </div>
                            <div className="favoriteCard__updateInfo">
                                <p className="updateInfo">
                                    Last Update: {Math.round((new Date() - new Date(joke.updated_at)) / 1000 / 60 / 60)}   &nbsp;hours ago
                                </p>
                            </div>
                        </div>
                    </div>
                ))
            }
        </Fragment>
    )
};
export default FavoriteCardBluePrint;