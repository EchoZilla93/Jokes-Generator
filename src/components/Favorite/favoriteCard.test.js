import React from 'react';
import FavoriteCardBluePrint from './FavoriteCardBluePrint';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import checkPropTypes from 'check-prop-types';
const setUp = (props={}) => {
    const component = shallow(<FavoriteCardBluePrint{...props}/>);
    return component;
}

describe('favorite card component',() => {
    describe('Cheking propTypes', () => {
        it('Should not throw an error', () => {
            const expectedProps = {
                id:'string',
                text:'string',
                updateInfo:0
            };
            const propsErr = checkPropTypes(FavoriteCardBluePrint.propTypes,expectedProps,'props',FavoriteCardBluePrint.name);
            expect(propsErr).toBeUndefined();
        })
    })
    let component;
    beforeEach(() => {
        component = setUp();
    });
    it('renders expected snapshot',() => {
        const wrapper = component;
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})