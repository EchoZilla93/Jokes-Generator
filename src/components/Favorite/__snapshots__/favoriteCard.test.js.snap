// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorite card component renders expected snapshot 1`] = `
<div>
  <div
    className="favoriteCard"
  >
    <div
      className="favoriteCard__status"
    >
      <i
        className="far fa-heart heart"
      />
    </div>
    <div
      className="favoriteCard__staticImage"
    >
      <i
        className="far fa-comment-dots static"
      />
    </div>
    <div
      className="favoriteCard__mainContent"
    >
      <h6
        className="favoriteCard__ID"
      >
        ID: 
        <a
          href="/"
        >
          helloWorld
        </a>
      </h6>
      <div
        className="textcontent"
      >
        <p>
          Chuckies joke
        </p>
      </div>
    </div>
    <div
      className="favoriteCard__updateInfo"
    >
      <p
        className="updateInfo"
      >
        Last Update: 200 hours ago
      </p>
    </div>
  </div>
</div>
`;
